packages/core/src/fhir-to-cda/cda-templates/components/observations.ts

```diff
@@ -0,0 +1,57 @@
+import { Observation } from "@medplum/fhirtypes";
+import { CDAInstanceIdentifier, CDACodeCV, Entry } from "../types";
+import {
+  buildInstanceIdentifiersFromIdentifier,
+  buildCodeCVFromCodeableConcept,
+  buildInstanceIdentifier,
+  withoutNullFlavorString,
+} from "../commons";
+import { classCodeAttribute, moodCodeAttribute } from "../constants";
+
+export interface CDAObservation {
+  component: {
+    observation: {
+      [classCodeAttribute]: Entry;
+      [moodCodeAttribute]: Entry;
+      id?: CDAInstanceIdentifier[] | Entry;
+      code: CDACodeCV | Entry;
+      text?: Entry;
+      statusCode?: {
+        code: Entry;
+      };
+      effectiveTime?: {
+        value: Entry;
+      };
+      priorityCode?: Entry;
+      // TODO support other types of values like CodeableConcept, Quantity, etc.
+      value?: string | undefined;
+    };
+  };
+}
+
+export function buildObservations(observations: Observation[]): CDAObservation[] {
+  return observations.map(observation => {
+    const effectiveTime = observation.effectiveDateTime?.replace(/-|:|\.\d+Z$/g, "");
```

Comment:
regex as an external constant so its not compiled on each call, please - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#63afb0983d864b61ab907a44801f72e1).

Reply:
Also fixing this in my follow up PR